iris-mpc-cpu/src/hawkers/galois_store.rs

```diff
@@ -184,43 +275,29 @@ impl VectorStore for LocalNetAby3NgStoreProtocol {
         distance1: &Self::DistanceRef,
         distance2: &Self::DistanceRef,
     ) -> bool {
-        let d1 = *distance1;
-        let d2 = *distance2;
         let ready_sessions = self.runtime.create_player_sessions().await.unwrap();
         let mut jobs = JoinSet::new();
-        for player in self.runtime.identities.clone() {
-            let mut player_session = ready_sessions.get(&player).unwrap().clone();
-            let storage = self.players.get(&player).unwrap();
-            let (x1, y1) = (
-                storage.points[d1.0.val()].clone(),
-                storage.points[d1.1.val()].clone(),
-            );
-            let (x2, y2) = (
-                storage.points[d2.0.val()].clone(),
-                storage.points[d2.1.val()].clone(),
-            );
-            let mut pairs = [(x1.data, y1.data), (x2.data, y2.data)];
-            jobs.spawn(async move {
-                pairs.iter_mut().for_each(|(_x, y)| {
-                    y.code.preprocess_iris_code_query_share();
-                    y.mask.preprocess_mask_code_query_share();
-                });
-                let ds_and_ts = galois_ring_pairwise_distance(&mut player_session, &pairs)
-                    .await
-                    .unwrap();
-                let ds_and_ts = galois_ring_to_rep3(&mut player_session, ds_and_ts)
-                    .await
-                    .unwrap();
-                cross_compare(
-                    &mut player_session,
-                    ds_and_ts[0].clone(),
-                    ds_and_ts[1].clone(),
-                    ds_and_ts[2].clone(),
-                    ds_and_ts[3].clone(),
-                )
-                .await
-                .unwrap()
-            });
+        for share1 in distance1.iter() {
+            for share2 in distance2.iter() {
+                if share1.player == share2.player {
```

Comment:
why is this needed?

Reply:
We want to be sure that these share belong to the same player before doing local operations.

Reply:
I see. The code will change a bit after I'll finish up the session refactoring. Nevertheless, that's a problem for future us :smile: